{
  ClassNode classNode=new ClassNode(Type.makeType("Foo"),ACC_PUBLIC,Type.OBJECT_TYPE);
  classNode.addConstructor(new ConstructorNode(ACC_PUBLIC,null));
  Parameter[] parameters={new Parameter(Type.OBJECT_TYPE.makeArray(),"coll")};
  Statement loopStatement=createPrintlnStatement(new VariableExpression("i"));
  ForStatement statement=new ForStatement("i",Type.DYNAMIC_TYPE,new VariableExpression("coll"),loopStatement);
  classNode.addMethod(new MethodNode("iterateDemo",ACC_PUBLIC,Type.VOID_TYPE,parameters,statement));
  Class fooClass=loadClass(classNode);
  assertTrue("Loaded a new class",fooClass != null);
  Object bean=fooClass.newInstance();
  assertTrue("Managed to create bean",bean != null);
  System.out.println("################ Now about to invoke a method with looping");
  Object[] array={new Integer(1234),"abc","def"};
  try {
    InvokerHelper.invokeMethod(bean,"iterateDemo",new Object[]{array});
  }
 catch (  InvokerInvocationException e) {
    System.out.println("Caught: " + e.getCause());
    e.getCause().printStackTrace();
    fail("Should not have thrown an exception");
  }
  System.out.println("################ Done");
}
